wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+                self.bot.guilds[0].channels,
+                name=f"{self.config.get_config_value('basic_config', 'BRANCH_NAME').lower()}_mod_channel"
+            )
+            if self.mod_channel is None:
+                self.mod_channel = await self.bot.guilds[0].create_text_channel(
+                    f"{self.config.get_config_value('basic_config', 'BRANCH_NAME')}_mod_channel"
+                )
+                mod_channel_id = self.mod_channel.id
+                if mod_channel_id is None:
+                    logger.info(
+                        "[Ban load()] the channel designated for mod reports "
+                        f"[{self.config.get_config_value('basic_config', 'BRANCH_NAME')}_mod_channel] "
+                        f"in {self.config.get_config_value('basic_config', 'BRANCH_NAME')} "
+                        "does not exist and I was unable to create it, exiting now...."
+                    )
+                    exit(1)
+                logger.info("[Ban load()] variable "
+                            f"\"mod_channel_id\" is set to \"{mod_channel_id}\"")
+
+    @commands.Cog.listener(name='on_member_join')
+    async def watchdog(self, member: discord.Member):
+        """Watches for users joining the guild and kicks and bans a user if they are banned"""
+
+        if member.id in self.ban_list:
+            logger.info(f"[Ban watchdog()] banned member, {member}, detected. Promply will notify and kick them.")
+
+            e_obj = discord.Embed(title="Ban Notification",
+                                  color=discord.Color.red(),
+                                  timestamp=datetime.datetime.now(pytz.timezone('Canada/Pacific'))
+                                  )
+            e_obj.add_field(name="Notice", value=f"**You are PERMANENTLY BANNED from\n{self.bot.guilds[0]}\n\n"
+                            "You may NOT rejoin the guild!**")
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+
+            await member.send(embed=e_obj)
+            await member.kick(reason="Not allowed back on server.")
+
+    @commands.Cog.listener(name='on_member_ban')
+    async def intercept(self, guild: discord.Guild, member: Union[discord.User, discord.Member]):
+        """Watches for a guild ban. The guild ban is undone and the user is banned via this ban system"""
+
+        # need to read the audit log to grab mod, date, and reason
+        logger.info(f"[Ban intercept()] guild ban detected and intercepted for user='{member}'")
+        try:
+            audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+        except Exception as e:
+            logger.info(f'error while fetching ban data: {e}')
+            await self.mod_channel.send(f"Encountered following error while intercepting a ban: {e}\n" +
+                                        "**Most likely need view audit log perms.**")
+            return
+        audit_ban = audit_ban[0]
```

Comment:
the idea here being that this function only gets called when a moderator does a guild ban and therefore the guiild ban is the latest action perform that is relevant?

in which case, why not do 
`audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()[0]`?

Reply:
u cant index a coroutine

Reply:
that line isn't gonna error so i didnt include it in the try block